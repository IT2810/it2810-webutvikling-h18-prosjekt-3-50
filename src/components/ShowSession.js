import React, { Component } from 'react'
import { Card, Text, CardItem, Body, Right, Left } from 'native-base'
import Moment from 'react-moment'

class ShowSession extends Component {

  constructor(props, context) {
    super(props, context)

    this.getSession = this.getSession.bind(this)
    this.getSelectedDate = this.getSelectedDate.bind(this)
    this.getDateText = this.getDateText.bind(this)
    this.getTimeCount = this.getTimeCount.bind(this)
  }

  getSelectedDate() {
    // Get selected date from store
    return new Date()
  }

  getSession(date) {
    // TODO: get from redux
    return {
      name: "Test Exercise",
      dateTime: new Date(2018, 9, 2, 16, 30, 0, 0),
      contacts: ["Kari Nordmann", "PÃ¥l Hansen", "Line Kristiansen"],
      exercises: [
        {name: 'Squat', sets: '4', reps: '12'},
        {name: 'Benchpress', sets: '4', reps: '12'},
        {name: 'Pullup', sets: '4', reps: '12'},
        {name: 'Row', sets: '4', reps: '12'}
      ]
    }
  }

  getDateText(session) {
    let today = new Date().getDate()
    if (session == null) {
      return <Text>No session planned for today </Text>
    } else if (session.dateTime.getDate() == today) {
      return <Text>Todays session</Text>
    } else {
      return <Text>
        <Text>Date: </Text>
        <Moment element={Text} format="D. MMMM">
          {session.dateTime}
        </Moment>
      </Text>
    }

  }

  getTimeCount = (dateTime) => (
    <Moment element={Text} fromNow={new Date() > dateTime} toNow={new Date() <= dateTime}>
      {dateTime}
    </Moment>
  )

  render() {
    const date = this.getSelectedDate()
    const session = this.getSession(date)
    const trainingPartners = session.contacts.join(', ')
    const exercises = session.exercises.map(exercise => exercise.name).join(', ')

    console.log("This is the session")
    console.log(session)

    const dateText = this.getDateText(session)

    const timeCount = this.getTimeCount(session.dateTime)

    return (
      <Card>
        <CardItem header bordered>
          <Body>
              {dateText}
          </Body>
          <Right>
            <Text> ( {timeCount} ) </Text>
          </Right>
        </CardItem>
        <CardItem bordered>
            <Text>
              <Text>Time: </Text>
              <Moment element={Text} format="HH:mm">
                {session.dateTime}
              </Moment>
            </Text>
        </CardItem>
        <CardItem bordered>
          <Text>Title: {session.name} </Text>
        </CardItem>
        <CardItem bordered>
          <Text>
            With: {trainingPartners}
          </Text>
        </CardItem>
        <CardItem bordered>
          <Text>
            Exercises: {exercises}
          </Text>
        </CardItem>
      </Card>
    )
  }

}

export default ShowSession
