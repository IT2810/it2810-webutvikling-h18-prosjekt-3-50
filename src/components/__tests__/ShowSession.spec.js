/**
 * @jest-environment jsdom
 */

import React from 'react'
import { shallow } from 'enzyme'
import ShowSession from '../ShowSession'
import toJson from 'enzyme-to-json'
import Moment from 'react-moment'
import { Card, Text, CardItem, Body, Right, Left } from 'native-base'

describe('ShowSession', () => {
  let wrapper
  let navigateMock

  beforeEach(() => {
    navigateMock = jest.fn()
    const navigation = { navigate: navigateMock }
    wrapper = shallow(<ShowSession navigation={navigation} />)
  })

  it('renders correctly', () => {
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

  describe('getDateText', () => {
    it('if no session is planned for the date', () => {
      let text = wrapper.instance().getDateText(null)

      expect(text).toEqual(
        <Text>No session planned for today </Text>
      )
    })

    it('if there is a session planned for the date', () => {
      let session = { dateTime: new Date() }

      let text = wrapper.instance().getDateText(session)

      expect(text).toEqual(
        <Text>Todays session</Text>
      )
    })

    it('if the session is for a date that was', () => {
      let session = { dateTime: new Date(2018, 10, 4) }

      let text = wrapper.instance().getDateText(session)

      expect(text).toEqual(
        <Text> Session the
          <Moment element={Text} format="DD.MM">
            {session.dateTime}
          </Moment>
        </Text>
      )
    })
  })

  describe('getTimeCount', () => {
    it('if date is before now', () => {
      let today = new Date()
      let dateTime = today.setDate(today.getDate() - 1)

      let text = wrapper.instance().getTimeCount(dateTime)

      expect(text).toEqual(
        <Moment element={Text} toNow>
          {dateTime}
        </Moment>
      )
    })

    it('if date is after now', () => {
      let today = new Date()
      let dateTime = today.setDate(today.getDate() + 1)

      let text = wrapper.instance().getTimeCount(dateTime)

      expect(text).toEqual(
        <Moment element={Text} fromNow>
          {dateTime}
        </Moment>
      )
    })
  })
})
