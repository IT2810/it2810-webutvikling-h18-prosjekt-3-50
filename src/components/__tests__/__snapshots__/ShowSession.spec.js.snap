// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowSession renders correctly 1`] = `
<Card
  style={
    Object {
      "backgroundColor": "#fff",
      "borderBottomWidth": 0.5,
      "borderColor": "#ccc",
      "borderLeftWidth": 0.5,
      "borderRadius": 2,
      "borderRightWidth": 0.5,
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexWrap": "nowrap",
      "marginBottom": 5,
      "marginLeft": 2,
      "marginRight": 2,
      "marginTop": 5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 1.5,
    }
  }
>
  <Styled(CardItem)
    bordered={true}
    header={true}
  >
    <Styled(Text)>
       Session the
      <t
        ago={false}
        calendar={false}
        decimal={false}
        element={[Function]}
        filter={[Function]}
        format="DD.MM"
        fromNow={false}
        interval={60000}
        onChange={[Function]}
        titleFormat=""
        toNow={false}
        unit={null}
        unix={false}
        utc={false}
        withTitle={false}
      >
        <Component />
      </t>
    </Styled(Text)>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <Styled(Text)>
      Title: 
      Test Exercise
       
    </Styled(Text)>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <Styled(Body)>
      <Styled(Text)>
        Time:
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={[Function]}
          filter={[Function]}
          format="HH:mm"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
      </Styled(Text)>
    </Styled(Body)>
    <Styled(Right)>
      <Styled(Text)>
        (
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element={[Function]}
          filter={[Function]}
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          titleFormat=""
          toNow={true}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          <Component />
        </t>
        )
      </Styled(Text)>
    </Styled(Right)>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <Styled(Text)>
      With: 
      Kari Nordmann, PÃ¥l Hansen, Line Kristiansen
    </Styled(Text)>
  </Styled(CardItem)>
  <Styled(CardItem)
    bordered={true}
  >
    <Styled(Text)>
      Exercises: 
      Squat, Benchpress, Pullup, Row
    </Styled(Text)>
  </Styled(CardItem)>
</Card>
`;
